site/snippets/signers/alchemy/components/EmailAuth.tsx

```diff
@@ -0,0 +1,68 @@
+import { AlchemySigner } from "@alchemy/aa-alchemy";
+import { useMutation } from "@tanstack/react-query";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useCallback, useMemo } from "react";
+
+export const LoginComponent = () => {
+  // Assumes that you are using Next.js but you can use any other router library
+  const params = useSearchParams();
+  const router = useRouter();
+
+  // It is recommended you wrap this in React Context or other state management
+  // For this example, we are defining the Alchemy Signer in the component
+  const signer = useMemo(
+    () =>
+      new AlchemySigner({
+        client: {
+          connection: {
+            jwt: "alcht_<KEY>",
+          },
+          iframeConfig: {
+            iframeContainerId: "turnkey-iframe-container",
+          },
+        },
+      }),
+    []
+  );
+
+  // we are using react-query to handle loading states more easily,
+  // feel free to use whichever state management library you prefer
+  const authUser = useCallback(async () => {
```

Comment:
this isn't quite right. you still have to initiate login with `signer.authenticate`

Reply:
we should merge this component with the one below. we need to handle the `bundle` as well